### ISSUE TITLE
No such table error

### ISSUE THREAD

**Issue by asteriscos** [wazuh/wazuh#24049]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.9.0 | Wazuh component | Manager/Agent | Sources | Ubuntu 20.04 |

Whenever we register several agents at the same time, with a docker environment, sometimes we see missing tables errors that can cause issues when Wazuh dashboard tries to access certain information.

<details>
<summary>Logs</summary>

```
Jun 11, 2024 @ 13:56:13.000 wazuh-db ERROR  DB(002) sqlite3_prepare_v2() stmt(64): no such table: sca_scan_info
Jun 11, 2024 @ 13:56:13.000 wazuh-analysisd ERROR  Error querying policy monitoring database for agent '002'
Jun 11, 2024 @ 13:56:13.000 wazuh-db ERROR  DB(002) sqlite3_prepare_v2() stmt(66): no such table: sca_policy
Jun 11, 2024 @ 13:56:13.000 wazuh-analysisd ERROR  Error querying policy monitoring database for agent '002'
Jun 11, 2024 @ 13:56:13.000 wazuh-db ERROR  DB(002) sqlite3_prepare_v2() stmt(69): no such table: sca_check
Jun 11, 2024 @ 13:56:13.000 wazuh-analysisd ERROR  Error querying policy monitoring database for agent '002'
Jun 11, 2024 @ 13:56:13.000 wazuh-db ERROR  DB(003) sqlite3_prepare_v2() stmt(64): no such table: sca_scan_info
Jun 11, 2024 @ 13:56:13.000 wazuh-analysisd ERROR  Error querying policy monitoring database for agent '003'
Jun 11, 2024 @ 13:56:13.000 wazuh-db ERROR  DB(003) sqlite3_prepare_v2() stmt(66): no such table: sca_policy
Jun 11, 2024 @ 13:56:13.000 wazuh-analysisd ERROR  Error querying policy monitoring database for agent '003'
Jun 11, 2024 @ 13:56:13.000 wazuh-db ERROR  DB(003) sqlite3_prepare_v2() stmt(69): no such table: sca_check
Jun 11, 2024 @ 13:56:13.000 wazuh-analysisd ERROR  Error querying policy monitoring database for agent '003'
Jun 11, 2024 @ 13:56:13.000 wazuh-db ERROR  DB(002) sqlite3_prepare_v2() stmt(70): no such table: sca_policy
Jun 11, 2024 @ 13:56:13.000 wazuh-analysisd ERROR  Error querying policy monitoring database for agent '002'
Jun 11, 2024 @ 13:56:13.000 wazuh-db ERROR  DB(003) sqlite3_prepare_v2() stmt(70): no such table: sca_policy
Jun 11, 2024 @ 13:56:13.000 wazuh-analysisd ERROR  Error querying policy monitoring database for agent '003'
Jun 11, 2024 @ 13:56:13.000 wazuh-db ERROR  DB(003) sqlite3_prepare_v2() stmt(45): no such table: scan_info
Jun 11, 2024 @ 13:56:13.000 wazuh-db ERROR  DB(002) sqlite3_prepare_v2() stmt(45): no such table: scan_info
Jun 11, 2024 @ 13:56:13.000 wazuh-db ERROR  DB(003) Cannot cache statement
Jun 11, 2024 @ 13:56:13.000 wazuh-db ERROR  DB(002) Cannot cache statement
Jun 11, 2024 @ 13:56:13.000 wazuh-analysisd ERROR  FIM decoder: Bad response from database: Cannot save fim control message
Jun 11, 2024 @ 13:56:13.000 wazuh-analysisd ERROR  FIM decoder: Bad response from database: Cannot save fim control message
Jun 11, 2024 @ 13:56:13.000 wazuh-db ERROR  DB(003) sqlite3_prepare_v2() stmt(46): no such table: scan_info
Jun 11, 2024 @ 13:56:13.000 wazuh-db ERROR  DB(003) Cannot cache statement
```

</details>

<details>
<summary>UI errors</summary>

![image](https://github.com/wazuh/wazuh/assets/9343732/7ec65b52-3550-4361-8157-74d1f4793118)

![image](https://github.com/wazuh/wazuh/assets/9343732/271e5c58-67bf-4bd2-abbc-4ac0e39563ca)

</details>

<details>
<summary>Agents tables</summary>

```

root@wazuh-manager-4:/var/ossec/queue/db# sqlite3 001.db
sqlite> .tables
fim_entry     sys_hotfixes  sys_netaddr   sys_netproto  sys_ports   
pm_event      sys_hwinfo    sys_netiface  sys_osinfo    sys_programs
sqlite> ^C

root@wazuh-manager-4:/var/ossec/queue/db# sqlite3 002.db
SQLite version 3.37.2 2022-01-06 13:25:41
Enter ".help" for usage hints.
sqlite> .tables
fim_entry     sys_hotfixes  sys_netaddr   sys_netproto  sys_ports   
pm_event      sys_hwinfo    sys_netiface  sys_osinfo    sys_programs
sqlite> ^C

root@wazuh-manager-4:/var/ossec/queue/db# sqlite3 003.db
SQLite version 3.37.2 2022-01-06 13:25:41
Enter ".help" for usage hints.
sqlite> .tables
ciscat_results        sca_policy            sys_netiface        
fim_entry             sca_scan_info         sys_netproto        
metadata              scan_info             sys_osinfo          
pm_event              sync_info             sys_ports           
sca_check             sys_hotfixes          sys_processes       
sca_check_compliance  sys_hwinfo            sys_programs        
sca_check_rules       sys_netaddr         
sqlite> 
```

</details>

### Example docker-compose

[docker-compose.zip](https://github.com/user-attachments/files/15789979/docker-compose.zip)

**Comment by vikman90** [comment]
## Related issues

The following issues were fixed in 4.7.5:

- #22847
  - #23467
- #23623
  - #23657

### Upward merge

On the other hand, the fix is in branch 4.9.0 as of May 31:

- #23819

### Next steps

Please confirm that the Docker image was built after May 31.

**Comment by vikman90** [comment]
**Update:** this seems to be a false alarm.
